redash/query_runner/yandex_metrica.py

```diff
@@ -120,18 +125,25 @@ def _get_tables(self, schema):
         return list(schema.values())
 
     def test_connection(self):
-        self._send_query("management/v1/{0}".format(self.list_path))
+        self._send_query(f"management/v1/{self.list_path}")
 
+    @backoff.on_exception(backoff.fibo, QuotaException, max_tries=10)
     def _send_query(self, path="stat/v1/data", **kwargs):
         token = kwargs.pop("oauth_token", self.configuration["token"])
         r = requests.get(
-            "{0}/{1}".format(self.url, path),
-            headers={"Authorization": "OAuth {}".format(token)},
+            f"{self.url}/{path}",
+            headers={"Authorization": f"OAuth {token}"},
             params=kwargs,
         )
-        if r.status_code != 200:
-            raise Exception(r.text)
-        return r.json()
+
+        response_data = r.json()
+
+        if not r.ok:
+            error_message = f"Code: {r.status_code}, message: {r.text}"
+            if response_data["code"] == 429:
```

Comment:
@denisov-vlad shouldn't this be `if r.status_code == 429`? Can you provide an example 429 response from Metrica?

Reply:
@guidopetri thanks! Yes, `r.status_code` is better here.